trusted_applet/net.go

```diff
@@ -77,6 +82,56 @@ func init() {
 	})
 }
 
+// getHttpClient returns a http.Client instance which uses the local resolver.
+func getHttpClient() *http.Client {
+	netTransport := &http.Transport{
+		DialContext: func(_ context.Context, network, add string) (net.Conn, error) {
+			glog.V(2).Infof("Resolving IP to dial %v", add)
+			parts := strings.Split(add, ":")
+			if len(parts) != 2 {
+				// Dial is only called with the host:port (no scheme, no path)
+				return nil, fmt.Errorf("expected host:port but got %q", add)
+			}
+			host, port := parts[0], parts[1]
+			// Look up the hostname
+			r, _, err := resolve(host)
+			if err != nil {
+				return nil, fmt.Errorf("failed to resolve host %q: %v", host, err)
+			}
+			if len(r.Answer) == 0 {
+				return nil, fmt.Errorf("failed to resolve A records for host %q", host)
+			}
+			var ip net.IP
+			for _, ans := range r.Answer {
+				if a, ok := ans.(*dns.A); ok {
```

Comment:
There could be multiple A records. I think it's quite debatable whether you should just stick to the first one, or find another one. Having said one, if the first one doesn't work, you'll want to try a different one.

Reply:
This is a good question, thank you!
I'm not sure whether client-side retying is the right answer (it's definitely what browsers do, but I believe it also comes with some downsides in that context which may/may not apply here).

For now I've just made it select a random A record from the returned list, at least that should enable us to make progress if the IP addresses pointed to by the first records don't work and the server isn't doing round-robin/record-rotation.